src/Stratis.Bitcoin/P2P/Peer/NetworkPeerConnection.cs

```diff
@@ -267,6 +260,7 @@ public async Task SendAsync(Payload payload, CancellationToken cancellation = de
                     byte[] bytes = ms.ToArray();
 
                     await this.SendAsync(bytes, cancellation).ConfigureAwait(false);
+                    this.asyncProvider.Signals.Publish(new PeerMessageSent(this.peer.PeerEndPoint, message, bytes.Length));
```

Comment:
I realize that this change will create a new instance of the message bug hub even if no one is listening.
Every time a message will be sent (or received in other cases) we create an instance of `PeerMessageSent`

Reply:
I wonder if we should keep regular events for none happy path.

Reply:
I'm aware of this and I'm not worried considering the number of events, I consider this a micro optimization that could be done if we ever see performance impact on this aspect (and must be proved by numbers) e.g. using object pooling or other tecniques.

P.S.
can't really compare this with regular events and if you want to mimic its behavior in terms of preserving instance creation, you'd have to check if the eventbus have any subscriber to that event before publishing it and I don't see a nice way without having to call something prior to publish the event and complicate the design... I've an idea tho that I want to experiment with (but as said this will complicate the event bus design a bit)

Reply:
Our node already consumes excessive memory this will only add on the burden and most of the time no one will subscribe to this event.

Reply:
consider that the PeerMessageSent class just takes few bytes per messages and got gargabe collected at gen0 if no one consumes it (the message itself if not stored goes out of scope and collected)
if number proves me wrong, will be easy to wrap this call around a property that reflects an argument like -publishpeerevents or something like that